src/Service/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -194,5 +197,59 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
 
             return sessionMapQuery.ToString();
         }
+
+        /// <inheritdoc/>
+        public override async Task<DbResultSet> GetMultipleResultSetsIfAnyAsync(
+            DbDataReader dbDataReader, List<string>? args = null)
+        {
+            // From the first result set, we get the count(0/1) of records with given PK.
+            DbResultSet resultSetWithCountOfRowsWithGivenPk = await ExtractResultSetFromDbDataReader(dbDataReader);
+            int numOfRecordsWithGivenPK = (int)resultSetWithCountOfRowsWithGivenPk.Rows.FirstOrDefault()!.Columns[MsSqlQueryBuilder.COUNT_ROWS_WITH_GIVEN_PK]!;
+
+            // The second result set holds the records returned as a result of the executed update/insert operation.
+            DbResultSet? dbResultSet = await dbDataReader.NextResultAsync() ? await ExtractResultSetFromDbDataReader(dbDataReader) : null;
+
+            if (dbResultSet is null)
+            {
+                // This can only happen in the case where we are dealing with table/view with auto-gen PK.
+                // In this case, insert cannot happen.
+                // Also since dbResultSet is null, update couldn't execute as well because there was no record corresponding to given PK.
```

Comment:
Here, why are we not checking the value of `numOfRecordsWithGivenPK` first before deciding what message to throw in the exception - either INSERT or update?

Reply:
As per discussion on call, explained better in the comment this scenario.

Reply:
Can you please explain why PK would be auto-gen in here? 

Is that because how we construct the Insert query for this case? Its hard to understand why this is so without cross referencing why this is - perhaps a line ref# would be helpful so code is more readable.